source/Missions/Services/Network/NetworkAgentRegistry.cs

```diff
@@ -64,11 +71,33 @@ private void Handle_AgentDeleted(MessagePayload<AgentDeleted> payload)
                 RemoveNetworkControlledAgent(agentId);
             }
         }
+        private bool RemoveNetworkControlledAgent(Guid agentId)
+        {
+            return m_OtherAgents.Values.Any(group =>
+            {
+                Agent agent = group.RemoveAgent(agentId);
+                if (agent != null)
+                {
+                    var result = m_AgentToId.Remove(agent);
+
+                    if (m_PlayerAgents.ContainsKey(agentId))
+                    {
+                        result &= m_PlayerAgents.Remove(agentId);
+                    }
```

Comment:
m_PlayerAgents.Remove(agentId) in this if will always return true, so we don't need to add it to the result.

We won't care about the result of `m_PlayerAgents.Remove(agentId)` due to this method's behavior being to primarily remove NetworkControlledAgent
```suggestion
                    RemovePlayerAgent(agentId);
```